src/app/ui/editor/brush_preview.cpp

```diff
@@ -512,13 +515,37 @@ void BrushPreview::invalidateRegion(const gfx::Region& region)
   m_clippingRegion.createSubtraction(m_clippingRegion, region);
 }
 
-void BrushPreview::generateBoundaries()
+void BrushPreview::calculateTileBoundariesOrigin(const doc::Grid& grid,
+                                                 const gfx::Point& spritePos)
+{
+  m_brushBoundaries.offset(-m_brushBoundaries.begin()[0].bounds().x,
+                           -m_brushBoundaries.begin()[0].bounds().y);
+  gfx::Point canvasPos = grid.tileToCanvas(grid.canvasToTile(spritePos));
+  m_brushBoundaries.offset(canvasPos.x - spritePos.x,
+                           canvasPos.y - spritePos.y);
+}
+
+void BrushPreview::generateBoundaries(Site &site, gfx::Point spritePos)
```

Comment:
To avoid a copy of the `Site` we can use a `const Site&`:
```suggestion
void BrushPreview::generateBoundaries(const Site& site,
                                      const gfx::Point& spritePos)
```

Reply:
Thanks! You added `const` to lock the values of the `Site` object inside the `generateBoundaries` function.
And you corrected `Site &site` as `Site& site` (I used to think `Site &site` is the same as `Site& site` @dacap could you confirm this?).

Reply:
Sorry my first comment about "to avoid a copy" was wrong for `Site` (it was for `gfx::Point`) as originally it was `Site& site`, not `Site site`. (When we use value parameters like `Site site` is when a copy is made, I read it wrong when wrote that comment.)

About `Site &site` vs `Site& site` there is no difference, it's just the formatting we are using and we have to keep it consistent.

Reply:
Oh! `gfx::Point`, I omitted to modify it. I pushed a new change to include `const gfx::Point&`.
Thanks for the clarification!